sequencer/src/lib.rs

```diff
@@ -286,38 +290,46 @@ pub async fn init_node(
     let public_ip = IpAddr::V4(Ipv4Addr::new(127, 0, 0, 1));
     let orchestrator_client = OrchestratorClient::new(validator_args, public_ip.to_string());
 
-    let (mut config, wait_for_orchestrator) = match persistence.load_config().await? {
+    let (config, wait_for_orchestrator) = match persistence.load_config().await? {
         Some(config) => {
             tracing::info!("loaded network config from storage, rejoining existing network");
             (config, false)
         }
         None => {
             tracing::info!("loading network config from orchestrator");
-            let config = orchestrator_client.get_config(public_ip.to_string()).await;
+            let mut config: NetworkConfig<VerKey, StaticElectionConfig> =
+                orchestrator_client.get_config(public_ip.to_string()).await;
+
+            config = orchestrator_client
+                .post_and_wait_all_public_keys(
+                    config.node_index,
+                    config.config.my_own_validator_config.public_key,
+                )
+                .await;
             tracing::info!("loaded config, we are node {}", config.node_index);
             persistence.save_config(&config).await?;
             (config, true)
         }
     };
     let node_index = config.node_index;
 
-    // Generate public keys and this node's private keys.
+    // Generate node's private keys.
     //
     // These are deterministic keys suitable *only* for testing and demo purposes.
```

Comment:
We can change this comment now. We're no longer generating the keys. And they are no longer unsafe (once we fix the above)